skema/text_reading/text_reading/Dockerfile

```diff
@@ -38,31 +38,44 @@ RUN sbt publishLocal
 # =============================================================================
 # CREATE A PYTHON VENV AND UPGRADE PYTHON TOOLS
 # =============================================================================
-ENV VIRTUAL_ENV=/opt/automates_venv
-RUN python3 -m venv $VIRTUAL_ENV
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+#ENV VIRTUAL_ENV=/opt/automates_venv
+#RUN python3 -m venv $VIRTUAL_ENV
+#ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+#
+#RUN pip install --upgrade setuptools
+#RUN pip install wheel
+# =============================================================================
+
+# =============================================================================
+# SETUP THE SKEMA TR ENVIRONMENT
+# =============================================================================
+RUN mkdir -p /skema/text_reading
+COPY . /skema/text_reading
+WORKDIR /skema/text_reading
+# Compile the fat jar
+RUN sbt assembly
 
-RUN pip install --upgrade setuptools
-RUN pip install wheel
 # =============================================================================
 
 # =============================================================================
-# SETUP THE AUTOMATES TR ENVIRONMENT
+# SETUP THE PYTHON ENVIRONMENT TO PROCESS THE PARQUET FILES
 # =============================================================================
-RUN mkdir -p /automates/automates/text_reading
-RUN mkdir -p /automates/automates/apps/pdfalign
-COPY automates/text_reading /automates/automates/text_reading
-COPY automates/apps/pdfalign /automates/automates/apps/pdfalign
-WORKDIR /automates/automates/text_reading
-RUN pip install -r requirements.txt
+RUN pip3 install pandas fastparquet tqdm
 
-# In order for the test suite to work, you need vector files. Run the following to get them:
-# RUN wget http://vanga.sista.arizona.edu/automates_data/vectors.txt -O /automates/automates/text_reading/src/test/resources/vectors.txt
-RUN sbt -Dapps.projectDir=/automates test
 # =============================================================================
+# PREPARE THE WORKING ENVIRONMENT
+# =============================================================================
+RUN mkdir -p /data/inputs
+RUN mkdir -p /data/json_inputs
+RUN mkdir -p /data/outputs
+RUN chmod 777 /data/json_inputs
+
+RUN chmod +x annotate_cosmos_json
+
+
 
 # =============================================================================
-# PREPARE FOR USE
+# ENTRY POINT IS THE  WEB API
 # =============================================================================
 CMD ["sbt", "webapp/run"]
```

Comment:
This will mean that the webapp is running in development mode, which might not be a good long-term solution.  I do not see, for example, how the $secret from application.conf is being managed.  Startup times for sbt can sometimes be frustrating.

Reply:
Indeed. If there is a way to run the web app without relying on `sbt`, we should call this instead.